underscore.js

```diff
@@ -520,19 +520,34 @@
     if (iteratee != null) iteratee = cb(iteratee, context);
     var result = [];
     var seen = [];
+    var seenPrimitive = {};
```

Comment:
Some browsers used to optimize for it. Doesn't seem necessary any more based on some benchmarks